### ISSUE TITLE
Loading bad program from command line stops cryptol from loading prelude

### ISSUE THREAD

**Issue by weaversa** [GaloisInc/cryptol#239]
```
$ cryptol broken.cry
                        _        _
   ___ _ __ _   _ _ __ | |_ ___ | |
  / __| '__| | | | '_ \| __/ _ \| |
 | (__| |  | |_| | |_) | || (_) | |
  \___|_|   \__, | .__/ \__\___/|_|
            |___/|_| version 2.3.0 (81b29cb)

Parse error at ./broken.cry:8:22, unexpected False
cryptol> [0..]

[error] at <interactive>:1:1--1:6:
  Type variable `bits` was not defined.
cryptol> 
```
versus
```
$ cryptol 
                        _        _
   ___ _ __ _   _ _ __ | |_ ___ | |
  / __| '__| | | | '_ \| __/ _ \| |
 | (__| |  | |_| | |_) | || (_) | |
  \___|_|   \__, | .__/ \__\___/|_|
            |___/|_| version 2.3.0 (81b29cb)

Loading module Cryptol
Cryptol> :l broken.cry

Parse error at ./broken.cry:8:22, unexpected False
cryptol> [0..]
Assuming a = 0
[0x0]
```

**Comment by elliottt** [comment]
This is a consequence of moving all the prims to the prelude.  It would be nice if we fell back to just loading the prelude if the focused module has an error in it.

**Comment by weaversa** [comment]
That sounds like a fine fix.

Also, is seems the types of the primitives are in the prelude, but not the definitions. Why break them up? What is the advantage of being able to specify the types in the prelude?

**Comment by elliottt** [comment]
We used to have one big Haskell module where the types of the primitive operations were constructed using Haskell functions.  While this did work, it meant that things like asking for the type of an expression or getting its documentation string had to consider two sources: a built-in primitive, or a user-defined function.

In order to make things a little more uniform, we introduced the primitive declarations.  Primitive declarations are just the type and name of the primitive functionality provided by the interpreter, to bootstrap typechecking.  This also motivated the documentation string feature, allowing us to move the documentation for all of the primitives out of another Haskell module, and into the Cryptol prelude.

However, if you introduce new primitives that lack implementations in the interpreter, your programs will typecheck, but they won't run, as the interpreter will panic when it hits the primitive that it doesn't have an implementation for :)

**Comment by brianhuffman** [comment]
I should point out another advantage of specifying types of primitives in the prelude: They can be written down using Cryptol's (readable) type syntax. When they were constructed from Haskell functions in the big Haskell module, they were very hard to read and it was hard to notice mistakes.

**Comment by robdockins** [comment]
Fixed via 21f5993730062ae0c1f392531b2f04824e503a2e
